### ISSUE TITLE
CHOLMOD supernodal factorization performance degradation due to OpenMP

### ISSUE THREAD

**Issue by harakas** [DrTimothyAldenDavis/SuiteSparse#34]
In my application I'm using the CHOLMOD Supernodal factorisation to solve sparse systems of somewhat small size (100-10000). I'm seeing excessive usage of CPU and performance degradation caused by OpenMP thread synchronisation.

My application with default settings runs for 161.27 seconds with average CPU usage of 250-300%. It does other things aside from solving systems, so probably only 50% of time is spent inside the solvers.

Once I disable OpenMP with `export OMP_THREAD_LIMIT=1`, my application runs for 155.98 seconds with average CPU usage of 99.9%.

I can also reduce general CPU usage with the `GOMP_SPINCOUNT=500` environment variable (to around 150%), however, this comes with a further increased run time.

I can see four threads taking CPU, with one at 99.9%, and the others at 50% and for the other 3 gdb always showing:
```
(gdb) bt
#0  0x00007fffa96ab112 in do_spin (val=1731152, addr=0x7fff3001d044) at /build/gcc/src/gcc/libgomp/config/linux/x86/futex.h:130
#1  0x00007fffa96ab112 in do_wait (val=1731152, addr=0x7fff3001d044) at /build/gcc/src/gcc/libgomp/config/linux/wait.h:66
#2  0x00007fffa96ab112 in gomp_barrier_wait_end (bar=0x7fff3001d040, state=1731152) at /build/gcc/src/gcc/libgomp/config/linux/bar.c:48
#3  0x00007fffa96a8a98 in gomp_simple_barrier_wait (bar=0x7fff3001d040) at /build/gcc/src/gcc/libgomp/config/posix/simple-bar.h:60
#4  0x00007fffa96a8a98 in gomp_thread_start (xdata=<optimized out>) at /build/gcc/src/gcc/libgomp/team.c:124
#5  0x00007ffff6d47a9d in start_thread () at /usr/lib/libpthread.so.0
#6  0x00007fffa95adb23 in clone () at /usr/lib/libc.so.6
(gdb) 
```
I'm using Arch Linux 1.4, SuiteSpare 5.3.0-1, GCC 7.4.1 and Eigen that wraps the CHOLMOD code. With Eigen's own sparse solver I'm seeing 99.9% CPU usage and run time of 175.2 seconds -- so clearly the problem is inside SuiteSparse.

**Comment by DrTimothyAldenDavis** [comment]
Which BLAS library are you using?

**Comment by harakas** [comment]
Package manager shows that I have blas 3.8.0-2 from lapack.

**Comment by DrTimothyAldenDavis** [comment]
I'm working on a beta version of CHOLMOD that should address this issue.  It's not yet tagged as a pre-release.  I need to do some more benchmarking.  The draft is on the dev2 branch of SuiteSparse

**Comment by DrTimothyAldenDavis** [comment]
This issue is likely fixed by https://github.com/OpenMathLib/OpenBLAS/pull/4441 which fixes  https://github.com/OpenMathLib/OpenBLAS/issues/2265 and https://github.com/OpenMathLib/OpenBLAS/issues/2846 .  The fix will appear in OpenBLAS 0.3.27.
